src/test/java/org/cactoos/iterable/FilteredTest.java

```diff
@@ -130,12 +138,63 @@ public void withItemsNotEmpty() {
     public void withoutItemsIsEmpty() {
         new Assertion<>(
             "Must be empty without items",
-            new Filtered<String>(
+            new Filtered<>(
                 input -> input.length() > 16,
                 new IterableOf<>("third", "fourth")
             ),
             new IsEmptyIterable<>()
         ).affirm();
     }
 
+    @Test
+    public void filtersWithFunc() {
+        new Assertion<>(
+            "Must be filtered with Scalar<Boolean> result",
+            new Filtered<>(
+                new IterableOf<>("a", "b", "c", "x"),
+                input -> new StartsWith(input, "c")
+            ),
+            new HasValues<>("c")
+        ).affirm();
+    }
+
+    @Test
+    public void filtersWithTwoFuncsCombined() throws Exception {
+        new Assertion<>(
+            "Must be filtered with two filters",
+            new Filtered<>(
+                new IterableOf<>("ay", "xb", "yx", "xy"),
+                FilteredTest.combine(
+                    input -> new StartsWith(input, "x"),
+                    input -> new EndsWith(input, "y")
```

Comment:
@andreoss what is test really testing? Clearly not `Filtered` since it is tested similarly in the previous method

Reply:
I was trying to emphasize the need for `func.And`, so I've added this test and the puzzle. 
I will remove the test above test than.

Reply:
@andreoss I think we should keep the other one that you deleted and remove all the extra constructors you added that are not needed and also to remove this combine method.The other test 1) was simpler 2) had less dependencies 3) perfectly did the job of testing `Filtered`.